### PR TITLE
Set process environment in Benchmark hook member methods

### PR DESCRIPTION
**Pull request by tristan0x**
Set environment in `pre_execute` and `post_execute` member methods
according to `environment` and `modules` execution settings.

### CODE REVIEW

**Comment by tristan0x** [comment]
At last, tests pass. Can you please review @ohm314 ?

**Comment by ohm314** [comment]
Awesome! Will review first thing when I get to the office

File: hpcbench/toolbox/environment_modules.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import os
+import subprocess
+
+from . process import find_executable
+
+
+MODULECMD = find_executable('modulecmd', required=False)
+
+
+def module(*args):
```
Comment (ohm314) [inline hpcbench/toolbox/environment_modules.py]: Maybe providing an interface to the purge, load, unload commands would be better. Why not just have a Module class. You can still provide a generic "command" method to catch all the rest?

File: hpcbench/driver.py [hunk 2]
```diff
@@ -1084,6 +1086,28 @@ def popen(self, stdout, stderr):
...
 
+    @contextlib.contextmanager
+    def module_env(self):
+        """Set current process environment according
+        to execution `environment` and `modules`
+        """
+        env = copy.copy(os.environ)
+        try:
+            for mod in self.execution.get('modules') or []:
+                module('load', mod)
```
Comment (ohm314) [inline hpcbench/driver.py]: it's really cool that we have now this module API but I think it's a pity that it doesn't express a little more (cf. below)

**Pull request merged**